ui/src/components/erc/ERC.js

```diff
@@ -17,14 +17,14 @@ import SubstitutionInfoPop from './Substitution/SubstitutionInfo';
 import Metadata from './Metadata/Metadata';
 import Shipment from './Shipment/Shipment';
 import { withRouter } from 'react-router-dom';
-import logo from '../../assets/img/DOI_logo.svg.png';
+import Doilogo from '../../assets/img/DOI_logo.svg.png';
 
 class ERC extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
             failure: false,
-            id: this.props.match.params.id,
+            id: this.props.id ? this.props.id: this.props.location.state ? this.props.location.state.id  ? this.props.location.state.id : this.props.match.params.id : this.props.match.params.id ,
```

Comment:
This line is really hard to read and understand. Can you add a comment what it does, or put it in a function with a speaking name?